docs/lesson-plans/week-7.md

```diff
@@ -0,0 +1,632 @@
+would be saved as "hello" and "hey" would be saved as "hey".
+
+
+##### CHAR vs VARCHAR
+
+The difference between `CHAR(x)` and `VARCHAR(x)` is that a `CHAR` will right-pad the entry with spaces if it isn't of 
+length `x`, but a `VARCHAR` would save the entered value. If either columns attempt to save a value that is longer than
+`x` characters, then the value will be truncated and an error or warning might be thrown.
+
+#### TEXT
+
+The `TEXT` type is a string of unlimited length. The text will not be truncated.
+
+##### VARCHAR/CHAR vs TEXT
+
+The difference between `CHAR(x)`/`VARCHAR(x)` and `TEXT` is that a `CHAR`/`VARCHAR` is limited by the length of `x`. If
+an entry is made with length greater than `x`, the value will be truncated and an error or warning might be thrown. 
+`TEXT` will not be truncated. It's generally more efficient to store and perform operations on a `CHAR` or `VARCHAR` 
+so if possible, try to use those. Columns like usernames, first/last names, emails, website URLs, and others, whose 
+length generally won't be longer than a certain length should be stored as that.
+
+#### INTEGER/BIGINTEGER
+
+The `INTEGER` or `INT` type is an integer with a range of [-2147483648, 2147483647]. The `BIGINTEGER`/`BIGINT`
+is an integer value with an extended range of [-9223372036854775808, 9223372036854775807] (wow that was terrible to type
+out). 
+
+#### DECIMAL
+
+The `DECIMAL` type is similar to a double/float in other programming languages. You can usually specify the precision
+and scale of the column by writing it like this: `DECIMAL(19, 2)`. 
+
+**Precision** defines the total count of significant digits in the number (total number of digits on *both sides* of the
+decimal point).
+**Scale** defines the count of digits to the right of the decimal.
+
+The range of this data type is *up to 131072 digits before the decimal point; up to 16383 digits after the decimal 
+point*.
+
+#### TIMESTAMP
+
+The `TIMESTAMP` type is one of the date/time types commonly seen in SQL databases. There's also a `TIMESTAMP` type with
+a time zone, if that suits your needs better. 
+
+#### BOOLEAN/TINYINTEGER
+
+The `BOOLEAN` is a type which is available in PostgreSQL. In many other DBs, `TINYINTEGER` or `TINYINT` is commonly used
+instead (they are the same thing). 
+
+#### ENUM
+
+The `ENUM` type allows you to specify a small set of possible values for a column. In PostgreSQL, enums are created as
+a new datatype and then used in tables afterward. In other DB management systems, enums are usually just written as 
+you're declaring a column type.
+
+Here's an example of creating an enum type, which is shown used in a table later on:
+```sql
+CREATE TYPE personality AS ENUM ('nice', 'mean', 'neither');
+```
+
+### Column Modifiers
+
+Column modifiers are special modifiers you add when declaring a column in a table which give it extra properties. Here
+are a few of the common ones (some that you may need soon too!).
+
+#### PRIMARY KEY
```

Comment:
include FOREIGN KEY too?

Reply:
I was going to include that in the following week, since I thought linking tables together might be a little too much for one week. Do you think it should be mentioned here?

Reply:
In that case, that's all good - doesn't need to be added here

Reply:
I kind of thing it should be mentioned here. I think this stuff is conceptually lighter than the content in most other weeks - worth lumping together.

Reply:
Got it, I'll throw it in then

Reply:
Let me know how it looks.